cmake/OpenCVDetectPython.cmake

```diff
@@ -26,7 +26,7 @@ function(find_python preferred_version min_version library_env include_dir_env
          libs_found libs_version_string libraries library debug_libraries
          debug_library include_path include_dir include_dir2 packages_path
          numpy_include_dirs numpy_version)
-if(NOT ${found})
+if(NOT ${found} OR ${executable})
```

Comment:
Why do you have defined `PYTHON2INTERP_FOUND` / `PYTHON3INTERP_FOUND` in your build environment? How?

Ensure to clear CMake cache in case of changes in build dependencies.

Reply:
These two variables existed in the original project and were not created in this modification. Their job is to keep track of whether the Python environment has been found.
The original logic of the project was that if a Python environment had already been found (usually the default base environment), the Python environment probe would not take place.
This is problematic and will not take effect if you manually specify Python rings as a configuration.
The reason for this problem is that the first time camke is configured, it matches Python in the default base environment, and the Python environment is found in the two variables mentioned above. Therefore, any subsequent Python environment specified by the user will not take effect.

Here I have optimized the logic to ensure that camke matches the default Python environment when first loaded, and does not affect users who do not need to specify the Python environment. Then add the Python compiler path. If it is not empty, the user may have made changes to the Python environment and need to probe the Python environment according to the configured path. This allows you to configure it as if it were a Python environment.

Reply:
> Why do you have defined `PYTHON2INTERP_FOUND` / `PYTHON3INTERP_FOUND` in your build environment? How?
> 
> Ensure to clear CMake cache in case of changes in build dependencies.

These two variables existed in the original project and were not created in this modification. Their job is to keep track of whether the Python environment has been found.
In addition, I have teased out the logic involved in the previous comment, please check it out~

Reply:
> This is problematic 

This is how "CMake cache" works.

---

> These two variables existed in the original project

Try to unset these conflicting variables before calling of OpenCV scripts.

---

> The reason for this problem is that the first time camke is configured

Avoid using of `cmake-gui`. It contradicts with "CMake cache" feature and doesn't really support well build dependencies reconfiguration. That confuses many users.
Use `cmake` and specify all required parameters from the command line. `cmake` should be called once.

Reply:
> > This is problematic
> 
> This is how "CMake cache" works.
> 
> > These two variables existed in the original project
> 
> Try to unset these conflicting variables before calling of OpenCV scripts.
> 
> > The reason for this problem is that the first time camke is configured
> 
> Avoid using of `cmake-gui`. It contradicts with "CMake cache" feature and doesn't really support well build dependencies reconfiguration. That confuses many users. Use `cmake` and specify all required parameters from the command line. `cmake` should be called once.



> > This is problematic
> 
> This is how "CMake cache" works.
> 
> > These two variables existed in the original project
> 
> Try to unset these conflicting variables before calling of OpenCV scripts.
> 
> > The reason for this problem is that the first time camke is configured
> 
> Avoid using of `cmake-gui`. It contradicts with "CMake cache" feature and doesn't really support well build dependencies reconfiguration. That confuses many users. Use `cmake` and specify all required parameters from the command line. `cmake` should be called once.

You are right that this problem only occurs when using` cmake-gui`. This change only fixes issues triggered when using `camke-gui` and does not affect using `cmake` directly. `camke-gui` is not the best choice, but it is widely used, especially for development on the Windows platform.

Reply:
> > This is problematic
> 
> This is how "CMake cache" works.
> 
> > These two variables existed in the original project
> 
> Try to unset these conflicting variables before calling of OpenCV scripts.
> 
> > The reason for this problem is that the first time camke is configured
> 
> Avoid using of `cmake-gui`. It contradicts with "CMake cache" feature and doesn't really support well build dependencies reconfiguration. That confuses many users. Use `cmake` and specify all required parameters from the command line. `cmake` should be called once.

You are right that this problem only occurs when using `cmake-gui`. This change only fixes issues triggered when using `camke-gui` and does not affect using `cmake` directly. `camke-gui` is not the best choice, but it is widely used, especially for development on the Windows platform.